tests/scripts/thread-cert/Cert_5_1_01_RouterAttach.py

```diff
@@ -72,112 +73,14 @@ def test(self):
-        msg.assertMleMessageContainsTlv(mle.TlvRequest)
-        msg.assertMleMessageDoesNotContainTlv(mle.AddressRegistration)
-
-        # 5 - Leader
-        msg = leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-        msg.assertSentToNode(self.nodes[ROUTER])
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.Address16)
-        msg.assertMleMessageContainsOptionalTlv(mle.NetworkData)
-        msg.assertMleMessageContainsOptionalTlv(mle.Route64)
-        msg.assertMleMessageContainsOptionalTlv(mle.AddressRegistration)
-
-        # 6 - Router
-        msg = router_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/as")
-        msg.assertCoapMessageContainsTlv(network_layer.MacExtendedAddress)
-        msg.assertCoapMessageContainsTlv(network_layer.Status)
-
-        # 7 - Leader
-        msg = leader_messages.next_coap_message("2.04")
-        msg.assertCoapMessageContainsTlv(network_layer.Status)
-        msg.assertCoapMessageContainsOptionalTlv(network_layer.RouterMask)
-
-        status_tlv = msg.get_coap_message_tlv(network_layer.Status)
-        self.assertEqual(network_layer.StatusValues.SUCCESS, status_tlv.status)
-
-        # 8 - Router
-        msg = router_messages.next_mle_message(mle.CommandType.LINK_REQUEST)
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.Challenge)
-        msg.assertMleMessageContainsTlv(mle.Version)
-        msg.assertMleMessageContainsTlv(mle.TlvRequest)
-
-        tlv_request = msg.get_mle_message_tlv(mle.TlvRequest)
-        self.assertIn(mle.TlvType.LINK_MARGIN, tlv_request.tlvs)
-
-        # 9 - Leader
-        msg = leader_messages.next_mle_message(mle.CommandType.LINK_ACCEPT_AND_REQUEST)
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.Response)
-        msg.assertMleMessageContainsTlv(mle.LinkLayerFrameCounter)
-        msg.assertMleMessageContainsTlv(mle.Version)
-        msg.assertMleMessageContainsTlv(mle.LinkMargin)
-        msg.assertMleMessageContainsOptionalTlv(mle.MleFrameCounter)
-        msg.assertMleMessageContainsOptionalTlv(mle.Challenge)
-
-        # 10 - Router
-        msg = router_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
-        msg.assertSentWithHopLimit(255)
-        msg.assertSentToDestinationAddress("ff02::1")
-        msg.assertMleMessageContainsTlv(mle.SourceAddress)
-        msg.assertMleMessageContainsTlv(mle.LeaderData)
-        msg.assertMleMessageContainsTlv(mle.Route64)
-
-        # 11 - Leader, Router1
         for addr in self.nodes[LEADER].get_addrs():
-            self.assertTrue(self.nodes[ROUTER].ping(addr))
+            if addr[0:4] == 'fe80':
```

Comment:
reuse `get_linklocal()` in node.py?

Reply:
use get_ip6_address instead to avoid spinel-cli unsupported ipaddr sub-cmd for linklocal